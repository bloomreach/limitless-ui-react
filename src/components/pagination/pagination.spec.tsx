import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import { describe, it, expect, vi, beforeEach } from 'vitest';

import { Pagination } from './';

describe('Pagination', () => {
  beforeEach(() => { cleanup(); });

  it('should call onPageChange when a page is clicked', () => {
    const onPageChange = vi.fn();
    const { getByText } = render(
      <Pagination.Root count={100} page={0} itemsPerPage={10} onPageChange={onPageChange}>
        <Pagination.Overview>
          <Pagination.ItemsPerPageSelector />
          <Pagination.Summary />
        </Pagination.Overview>
        <Pagination.Navigation>
          <Pagination.PreviousButton />
          <Pagination.Pages />
          <Pagination.NextButton />
        </Pagination.Navigation>
      </Pagination.Root>
    );

    fireEvent.click(getByText('2'));
    expect(onPageChange).toHaveBeenCalledWith(1);
  });

  it('should call onItemsPerPageChange when items per page is changed', () => {
    const onItemsPerPageChange = vi.fn();
    const { getByRole, getAllByRole } = render(
      <Pagination.Root
        count={100}
        page={1}
        itemsPerPage={10}
        itemsPerPageOptions={[10, 25, 50]}
        onItemsPerPageChange={onItemsPerPageChange}
      >
       <Pagination.Overview>
          <Pagination.ItemsPerPageSelector />
          <Pagination.Summary />
        </Pagination.Overview>
        <Pagination.Navigation>
          <Pagination.PreviousButton />
          <Pagination.Pages />
          <Pagination.NextButton />
        </Pagination.Navigation>
      </Pagination.Root>
    );

    fireEvent.click(getAllByRole('combobox')[0]);
    screen.getAllByRole('menuitem')[1].click();
    expect(onItemsPerPageChange).toHaveBeenCalledWith(25);
  });

  it('should render with different itemsPerPageOptions', () => {
    const { getAllByRole } = render(
      <Pagination.Root
        count={100}
        page={1}
        itemsPerPage={10}
        itemsPerPageOptions={[5, 10, 20]}
      >
        <Pagination.Overview>
          <Pagination.ItemsPerPageSelector />
          <Pagination.Summary />
        </Pagination.Overview>
        <Pagination.Navigation>
          <Pagination.PreviousButton />
          <Pagination.Pages />
          <Pagination.NextButton />
        </Pagination.Navigation>
      </Pagination.Root>
    );

    fireEvent.click(getAllByRole('combobox')[0]);
    const menuItems = screen.getAllByRole('menuitem');
    expect(menuItems).toHaveLength(3);
    expect(menuItems[0].textContent).toBe('5');
    expect(menuItems[1].textContent).toBe('10');
    expect(menuItems[2].textContent).toBe('20');
  });
});
