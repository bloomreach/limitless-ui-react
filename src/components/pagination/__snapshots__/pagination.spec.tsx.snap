// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > should render successfully and match snapshot 1`] = `
<div
  class="lui-pagination"
>
  <div
    class="lui-pagination-left"
  >
    <button
      aria-autocomplete="none"
      aria-controls="radix-:r0:"
      aria-expanded="false"
      class="lui-pagination-items-per-page-selector"
      data-state="closed"
      dir="ltr"
      role="combobox"
      type="button"
    >
      <span
        style="pointer-events: none;"
      >
        10
      </span>
      <svg
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <div
      class="lui-pagination-summary"
    >
      Showing 
      11
       
      to 
      20
       
      of 
      100
    </div>
  </div>
  <div
    aria-orientation="horizontal"
    data-orientation="horizontal"
    dir="ltr"
    role="toolbar"
    style="outline-color: none; outline-style: none; outline-width: initial;"
    tabindex="0"
  >
    <nav
      aria-label="Pagination Navigation"
      class="lui-pagination-navigation"
      role="navigation"
    >
      <ul>
        <li>
          <button
            aria-label="Go to previous page"
            class="lui-pagination-navigation-button"
            data-orientation="horizontal"
            data-radix-collection-item=""
            tabindex="-1"
            type="button"
          >
            <svg
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.84182 3.13514C9.04327 3.32401 9.05348 3.64042 8.86462 3.84188L5.43521 7.49991L8.86462 11.1579C9.05348 11.3594 9.04327 11.6758 8.84182 11.8647C8.64036 12.0535 8.32394 12.0433 8.13508 11.8419L4.38508 7.84188C4.20477 7.64955 4.20477 7.35027 4.38508 7.15794L8.13508 3.15794C8.32394 2.95648 8.64036 2.94628 8.84182 3.13514Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="0"
          >
            <button
              aria-label="Go to page 1"
              class="lui-pagination-navigation-button lui-pagination-navigation-item"
              data-orientation="horizontal"
              data-radix-collection-item=""
              tabindex="-1"
              type="button"
            >
              1
            </button>
          </div>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="-1"
          >
            <button
              aria-current="true"
              aria-label="Current page, page 2"
              class="lui-pagination-navigation-button lui-pagination-navigation-item lui-pagination-navigation-item-selected lui-pagination-navigation-item-disabled"
              data-orientation="horizontal"
              data-radix-collection-item=""
              disabled=""
              tabindex="-1"
              type="button"
            >
              2
            </button>
          </div>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="0"
          >
            <button
              aria-label="Go to page 3"
              class="lui-pagination-navigation-button lui-pagination-navigation-item"
              data-orientation="horizontal"
              data-radix-collection-item=""
              tabindex="-1"
              type="button"
            >
              3
            </button>
          </div>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="0"
          >
            <button
              aria-label="Go to page 4"
              class="lui-pagination-navigation-button lui-pagination-navigation-item"
              data-orientation="horizontal"
              data-radix-collection-item=""
              tabindex="-1"
              type="button"
            >
              4
            </button>
          </div>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="0"
          >
            <button
              aria-label="Go to page 5"
              class="lui-pagination-navigation-button lui-pagination-navigation-item"
              data-orientation="horizontal"
              data-radix-collection-item=""
              tabindex="-1"
              type="button"
            >
              5
            </button>
          </div>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="0"
          >
            <button
              aria-label="Go to page 6"
              class="lui-pagination-navigation-button lui-pagination-navigation-item"
              data-orientation="horizontal"
              data-radix-collection-item=""
              tabindex="-1"
              type="button"
            >
              6
            </button>
          </div>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="0"
          >
            <button
              aria-label="Go to page 7"
              class="lui-pagination-navigation-button lui-pagination-navigation-item"
              data-orientation="horizontal"
              data-radix-collection-item=""
              tabindex="-1"
              type="button"
            >
              7
            </button>
          </div>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="-1"
          >
            <button
              class="lui-pagination-navigation-button lui-pagination-navigation-item lui-pagination-navigation-item-disabled"
              data-orientation="horizontal"
              data-radix-collection-item=""
              disabled=""
              tabindex="-1"
              type="button"
            >
              ...
            </button>
          </div>
        </li>
        <li>
          <div
            aria-orientation="horizontal"
            data-orientation="horizontal"
            dir="ltr"
            role="toolbar"
            style="outline-color: none; outline-style: none; outline-width: initial;"
            tabindex="0"
          >
            <button
              aria-label="Go to page 10"
              class="lui-pagination-navigation-button lui-pagination-navigation-item"
              data-orientation="horizontal"
              data-radix-collection-item=""
              tabindex="-1"
              type="button"
            >
              10
            </button>
          </div>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="lui-pagination-navigation-button"
            data-orientation="horizontal"
            data-radix-collection-item=""
            tabindex="-1"
            type="button"
          >
            <svg
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6.1584 3.13508C6.35985 2.94621 6.67627 2.95642 6.86514 3.15788L10.6151 7.15788C10.7954 7.3502 10.7954 7.64949 10.6151 7.84182L6.86514 11.8418C6.67627 12.0433 6.35985 12.0535 6.1584 11.8646C5.95694 11.6757 5.94673 11.3593 6.1356 11.1579L9.565 7.49985L6.1356 3.84182C5.94673 3.64036 5.95694 3.32394 6.1584 3.13508Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
